src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/MarkAllNonSerializableFieldsTests.cs

```diff
@@ -379,7 +379,10 @@ End Class
         [Fact]
         public async Task CA2235WithArrayType()
         {
-            await VerifyCS.VerifyAnalyzerAsync(@"
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetFramework.Net472.Default,
```

Comment:
Does it mean that the test is not run under the netcoreapp3.1 context? If so wouldn't it be better to have a new property on the `AdditionalMetadataReferences` class with a value set conditionally to `Net472` or `NetCoreApp31`?

Reply:
This analyzer explicitly disables itself if the compilation does not target .NET Framework.

Reply:
> 
> This analyzer explicitly disables itself if the compilation does not target .NET Framework.

Sorry I left the comment on the first occurrence but this was more of a generic comment for your PR. Does all analyzers for which you have explicitly set `ReferenceAssemblies = ReferenceAssemblies.NetFramework.Net472.Default,` do not work for netcore?

Reply:
Yes, that is correct. In most cases (numerically), the APIs in question do not exist in .NET Standard 2.0. In other cases, the analyzer is disabled for that context.

Reply:
Thank you! I am/was surprised by the number of cases/analyzers not "working" for the netstandard part.